editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java

```diff
@@ -0,0 +1,350 @@
+      // open a dialog to edit the feature's attributes
+      showEditableAttributes(selectedFeature);
+    } else {
+      new Alert(Alert.AlertType.WARNING, "Feature of unexpected geometry type selected.").show();
+    }
+  }
+
+  /**
+   * Creates a dialog with text fields to allow editing of the given feature's 'AD_ADDRESS' and
+   * 'ST_STR_NAM' attributes.
+   *
+   * @param selectedFeature the feature to update
+   */
+  private void showEditableAttributes(Feature selectedFeature) {
+
+    // create a dialog to edit the attributes of the selected feature
+    EditAttributesDialog editAttributesDialog = new EditAttributesDialog(selectedFeature);
+
+    // show the dialog and wait for the user response
+    editAttributesDialog.showAndWait();
+
+    // update the selected feature's feature table
+    updateAttributes(selectedFeature);
+  }
+
+  /**
+   * Updates the attributes of the selected feature.
+   *
+   * @param selectedFeature the feature to update
+   */
+  static void updateAttributes(Feature selectedFeature) {
+
+    // update feature in the feature table
+    ListenableFuture<Void> editResultFuture = selectedFeature.getFeatureTable().updateFeatureAsync(selectedFeature);
+    editResultFuture.addDoneListener(() -> {
+        try {
+            editResultFuture.get();
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Error updating attributes.").show();
+        }
+      }
+    );
+  }
+
+  /**
+   * Moves the selected point feature to the given map point by updating the selected
+   * feature's geometry and feature table.
+   *
+   * @param mapPoint the location to move point feature
+   */
+  private void movePoint(Point mapPoint) {
+
+    // set the selected features' geometry to a new map point
+    selectedFeature.setGeometry(mapPoint);
+
+    // update the selected feature's feature table
+    updateAttributes(selectedFeature);
+
+    // clear selection of the point
+    clearSelection();
+  }
+
+  /**
+   * Moves the last of the vertex point of the currently selected polyline to the given map point by updating the
+   * selected feature's geometry and feature table.
+   *
+   * @param mapPoint the location to move polyline feature
+   */
+  private void movePolylineVertex(Point mapPoint) {
+
+    // get the selected feature's geometry as a polyline
+    Polyline polyline = (Polyline) selectedFeature.getGeometry();
+
+    // create a polyline builder to add and remove parts from the polyline
+    PolylineBuilder polylineBuilder = new PolylineBuilder(polyline);
+
+    // get the nearest vertex to the map point on the polyline
+    ProximityResult nearestVertex =
+      GeometryEngine.nearestVertex(polyline, (Point) GeometryEngine.project(mapPoint, polyline.getSpatialReference()));
+
+    // get the part of the polyline nearest to the map point
+    Part part = polylineBuilder.getParts().get((int) nearestVertex.getPartIndex());
```

Comment:
rather than casting this `long` to an `int` (which could result in a stack overflow if the `long` is greater than the maximum allowed `int` value), we can use the following method which would throw an informative exception in case of the above happening. Very unlikely in this sample but good practice I believe. 

```suggestion
    Part part = polylineBuilder.getParts().get(Math.toIntExact(nearestVertex.getPartIndex()));
```